doc/decisions/private_api.md

```diff
@@ -0,0 +1,50 @@
+# Private API
+
+## Problem
+
+Only `libelektra-core` is supposed to access private data, but this contradicts the goal to keep the library minimal.
+`kdbprivate.h` was too generic, it contained many other parts next to the struct definitions of Key/KeySet.
+
+## Constraints
+
+## Assumptions
+
+- The [C99 standard, section 5.2.4.1](http://www.open-std.org/jtc1/sc22/wg14/) gives following limit:
+  4095 external identifiers in one translation unit
+
+## Considered Alternatives
+
+- keep current situation
```

Comment:
shortly describe what you consider as "current situation" (to give contrast to your proposed solution).

Reply:
The "current situation" is hard to describe, would this be good enough:

```suggestion
- keep current situation: Theoretically there is `kdbprivate.h` and everything there is supposed to be private, but lots of code still uses it when it shouldn't. `kdb.h` is also the only header that is definitely public. All other headers are anybody's guess.
```